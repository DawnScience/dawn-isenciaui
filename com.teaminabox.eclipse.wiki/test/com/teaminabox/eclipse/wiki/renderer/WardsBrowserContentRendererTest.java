package com.teaminabox.eclipse.wiki.renderer;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import org.junit.*;

public final class WardsBrowserContentRendererTest extends AbstractContentRendererTest {

	protected AbstractContentRenderer getRenderer() {
		return new WardsBrowserContentRenderer();
	}

	@Test
	public void testIsHeaderWithHeader() {
		assertTrue(getRenderer().isHeader("'''foo'''"));
	}

	@Test
	public void testIsHeaderWithHeaderMarkupFollowedByText() {
		assertFalse(getRenderer().isHeader("'''foo'''" + " foo"));
	}

	@Test
	public void testIsHeaderWithEmphasis() {
		assertFalse(getRenderer().isHeader("'''''foo'''''"));
	}

	/**
	 * Test for [ 1089118 ] WardsWiki: markup in quote section is ignored.
	 */
	@Test
	public void testQuoteWithMarkup() {
		String markup = "\t :\t'''foo'''";
		String expected = "<p class=\"quote\"><b>foo</b></p>";
		assertRenderedContains(markup, expected);
	}

	@Test
	public void testWhiteSpacePreservedInMonospace() {
		String markup = " foo bar  ";
		String expected = "<pre class=\"monospace\"> foo bar  </pre>";
		assertRenderedContains(markup, expected);
	}

}
