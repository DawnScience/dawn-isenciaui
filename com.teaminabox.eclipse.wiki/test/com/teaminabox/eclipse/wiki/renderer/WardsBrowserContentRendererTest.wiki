Horizontal lines
----

* list
** list
*** list
** list

 some plain text
 and more


'''Header'''
'''''Not a header'''''

'''Quotations''' 
	 :	To be or not to be ...
	 
a line with WardsBrowserContentRendererTest''''''s plural, __emphasis__ and ''italic'' , another '''''emphasis''''' bit and WardsBrowserContentRendererTest''''''s plural

WardsBrowserContentRendererTest''''''s plural

|*A*|*Big*|*Table*|
|very|cool|indeed|
|WardsBrowserContentRendererTest|SnipSnapContentRendererTest|TwikiContentRendererTest|
|Acronym:WTF |http://www.google.com |EclipseWiki:license.html |
|and|images |http://www.google.com/images/logo.gif |
'''Images'''
http://www.google.com/images/logo.gif

Text with Java types com.teaminabox.foo.BigClass and an inner class com.teaminabox.foo.BigClass.InnerClass

Bug: [ 1490342 ] URL in brackets detected incorrectly: I am sure you have heard of Google (http://www.google.com).

Bug: [2014771] %-encoded URLs like this http://abc.def.se/WEBLINK/ViewDocs?DocumentName=EAB%2FOEP-07%3A0518&Latest=true

Embed:EmbeddedContent here

Embed:com.teaminabox.foo.BigClass

Eclipse:TEST_PROJECT_NAME/com/teaminabox/foo/BigClass.java should be a link
Eclipse:/com/teaminabox/foo/BigClass.java should be a link