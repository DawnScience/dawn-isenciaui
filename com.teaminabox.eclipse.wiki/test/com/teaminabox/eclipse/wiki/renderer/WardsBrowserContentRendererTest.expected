<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<html>
  <head>
    <title>Wards Browser Content Renderer Test</title>
<style type="text/css"><!--
body {
  text-align:justify;
  font: georgia, sans-serif;
  color: #000000;
  margin: 50px;
}

hr {
  width: 80%;
}

.monospace {
  font-family:monospace;
  padding: 0.5em;
  margin: 0 0 0.5em 0.5em;
  border: 1px solid #ccc;
  background: #eee;
}

.quote {
  padding: 0.5em;
  margin: 0 0 0.5em 0.5em;
  border: 1px solid #ccc;
  background: #f0f5fa;
}

h1,h2,h3,h4,h5,h6 {
  font-variant: small-caps;
  text-decoration: underline;
  color: #111;
}

li {
  text-align: left
}

table {
  border-spacing: 0px;
  border-collapse: collapse;
}

td,th {
  margin: 5px;
  padding: 5px;
  border:1px solid black;
}

th {
  font-variant: small-caps;
  background: #eee;
}

td {
  font: smaller;
}

a {
  text-decoration: none;
}

a:link { 
	color: #0000D0;
}
a:visited { 
	color: #0000D0;
}
a:hover { 
	color: #0000FF;
}
a:active { 
	color: #0000D0;
}

--></style>
  </head>
  <body>
<h1>Wards Browser Content Renderer Test</h1>
<p>
Horizontal lines</p>
<hr>
<p>
</p>
<ul><li>list</li>
<ul><li>list</li>
<ul><li>list</li>
</ul><li>list</li>
</ul></ul><p>
</p>
<pre class="monospace">
 some plain text
 and more</pre><p>
</p>
<p>
</p>
<p><strong>Header</strong></p>
<p>
<b><i>Not a header</i></b></p>
<p>
</p>
<p><strong>Quotations </strong></p>
<p class="quote">To be or not to be ...</p><p>
	 </p>
<p>
a line with <a href="http://--wiki/WardsBrowserContentRendererTest">WardsBrowserContentRendererTest</a>'s plural, <b>emphasis</b> and <i>italic</i> , another <b><i>emphasis</i></b> bit and <a href="http://--wiki/WardsBrowserContentRendererTest">WardsBrowserContentRendererTest</a>'s plural</p>
<p>
</p>
<p>
<a href="http://--wiki/WardsBrowserContentRendererTest">WardsBrowserContentRendererTest</a>'s plural</p>
<p>
</p>
<table ><tr><th>A</th><th>Big</th><th>Table</th></tr>
<tr><td>very</td><td>cool</td><td>indeed</td></tr>
<tr><td><a href="http://--wiki/WardsBrowserContentRendererTest">WardsBrowserContentRendererTest</a></td><td>SnipSnapContentRendererTest<a href="http://--wiki/SnipSnapContentRendererTest">?</a></td><td>TwikiContentRendererTest<a href="http://--wiki/TwikiContentRendererTest">?</a></td></tr>
<tr><td><a href="http://www.acronymfinder.com/af-query.asp?Acronym=WTF">Acronym:WTF</a> </td><td><a href="http://www.google.com">http://www.google.com</a> </td><td><a href="http://eclipsewiki.free.fr/index.php?page=license.html">EclipseWiki:license.html</a> </td></tr>
<tr><td>and</td><td>images </td><td><img alt="http://www.google.com/images/logo.gif" src="http://www.google.com/images/logo.gif"/> </td></tr>
</table>
<p><strong>Images</strong></p>
<p>
<img alt="http://www.google.com/images/logo.gif" src="http://www.google.com/images/logo.gif"/></p>
<p>
</p>
<p>
Text with Java types <a href="http://--wiki/Java:com.teaminabox.foo.BigClass">com.teaminabox.foo.BigClass</a> and an inner class <a href="http://--wiki/Java:com.teaminabox.foo.BigClass$InnerClass">com.teaminabox.foo.BigClass.InnerClass</a></p>
<p>
</p>
<p>
Bug: [ 1490342 ] URL in brackets detected incorrectly: I am sure you have heard of Google (<a href="http://www.google.com">http://www.google.com</a>).</p>
<p>
</p>
<p>
Bug: [2014771] %-encoded URLs like this <a href="http://abc.def.se/WEBLINK/ViewDocs?DocumentName=EAB%2FOEP-07%3A0518&amp;Latest=true">http://abc.def.se/WEBLINK/ViewDocs?DocumentName=EAB%2FOEP-07%3A0518&amp;Latest=true</a></p>
<p>
</p>
<p>
<p>
Embedded content with a link to <a href="http://--wiki/WardsBrowserContentRendererTest">WardsBrowserContentRendererTest</a></p>
 here</p>
<p>
</p>
<p>


<!-- ======================================================== -->
<!-- = Java Sourcecode to HTML automatically converted code = -->
<!-- =   Java2Html Converter V4.1 2004 by Markus Gebhard  markus@jave.de   = -->
<!-- =     Further information: http://www.java2html.de     = -->
<div align="left" class="java">
<table border="0" cellpadding="3" cellspacing="0" bgcolor="#ffffff">
   <tr>
  <!-- start source code -->
   <td nowrap="nowrap" valign="top" align="left">
    <code>
<font color="#808080">1</font>&nbsp;<font color="#7f0055"><b>package&nbsp;</b></font><font color="#000000">com.teaminabox.foo;</font><br/>
<font color="#808080">2</font>&nbsp;<font color="#7f0055"><b>public&nbsp;class&nbsp;</b></font><font color="#000000">BigClass&nbsp;</font><font color="#000000">{&nbsp;</font><font color="#7f0055"><b>class&nbsp;</b></font><font color="#000000">InnerClass&nbsp;</font><font color="#000000">{}&nbsp;}</font></code>
    
   </td>
  <!-- end source code -->
   </tr>
  <!-- start Java2Html link -->
   <tr>
    <td align="right">
<small>
<a href="http://www.java2html.de" target="_blank">Java2html</a>
</small>
    </td>
   </tr>
  <!-- end Java2Html link -->
</table>
</div>
<!-- =       END of automatically generated HTML code       = -->
<!-- ======================================================== -->

</p>
<p>
</p>
<p>
<a href="http://--wiki/Eclipse:TEST_PROJECT_NAME/com/teaminabox/foo/BigClass.java">Eclipse:TEST_PROJECT_NAME/com/teaminabox/foo/BigClass.java</a> should be a link</p>
<p>
<a href="http://--wiki/Eclipse:/com/teaminabox/foo/BigClass.java">Eclipse:/com/teaminabox/foo/BigClass.java</a> should be a link</p>
  </body>
</html>